pkg/buildkit/worker/pool.go

```diff
@@ -289,31 +292,40 @@ func (p *workerPool) buildEndpointURL(ctx context.Context, podName string) (stri
 	return u.String(), nil
 }
 
-// generates internal hostname for pod using endpoints
-func (p *workerPool) extractHostname(endpoints *corev1.Endpoints, podName string) (string, error) {
-	var hostname string
-
-scan:
-	for _, subset := range endpoints.Subsets {
-		for _, address := range subset.Addresses {
-			if address.TargetRef.Name == podName {
-				for _, port := range subset.Ports {
-					if port.Port == p.servicePort {
-						hostname = strings.Join([]string{address.Hostname, endpoints.Name, endpoints.Namespace}, ".")
-						p.log.Info("Found eligible endpoint address", "hostname", hostname)
-
-						break scan
-					}
-				}
-			}
+// generates internal hostname for pod using an endpoint slice
+func (p *workerPool) extractHostname(epSlice *discoveryv1.EndpointSlice, podName string) (handled bool, hostname string) {
```

Comment:
I don't totally understand why `hostname` can't just be checked against `""` instead of the additional `handled`?

Reply:
I mean, you're totally right, I just thought a boolean felt more definitive since it's a binary "yes" or "no". I guess in this case no is `""` and anything else is yes. Truthy strings from Ruby surface yet again! 😛